lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -1416,7 +1418,7 @@ static void emitMatchingRCAdjustmentsForCall(ApplyInst *Call, SILValue OnX) {
 
   // Emit a retain for the @owned return.
   SILBuilderWithScope Builder(Call);
-  Builder.createRetainValue(Call->getLoc(), OnX, Builder.getDefaultAtomicity());
+  OnX = Builder.emitCopyValueOperation(Call->getLoc(), OnX);
```

Comment:
One thing to be aware of is that when we compile without ownership, we return the retain_value's operand here.